### PR TITLE
Fix/default props function type lost

### LINKED ISSUE #913: Default props can't be `undefined` or `null`
### I am interested in helping provide a fix!

Yes

### Which generators are impacted?

- [ ] All
- [ ] Angular
- [ ] HTML
- [ ] Qwik
- [ ] React
- [ ] React-Native
- [ ] Solid
- [ ] Stencil
- [X] Svelte
- [ ] Vue
- [ ] Web components

### Reproduction case

https://mitosis.builder.io/?outputTab=M4NwpgNgLmQ%3D&code=JYWwDg9gTgLgBAbzgVwM4FMAi6BmBDZAGxgAUoIxU4BfOHckOAcgAEAjZYQgE3SgDpgEAPQhgMCKmComAbgBQ89AA9IsOL3xF4OZADsAxjCF64AWQCeAYQjgIe9HpgAKMOUoBKRPLgoM2LWIyClRnBB9fOD08EHQALhQ9TWAHbgAaCOoPBQiodBhkKFNnCN8AHm5gADcAPgQ3EP5o2Ooy4UraiOz5aiA

### Expected Behaviour

The output should be:
```svelte
<script>
  export let name = undefined;
</script>

<div>{name}</div>
```

### Actual Behaviour

```svelte
<script>
  export let name;
</script>

<div>{name}</div>
```

### Additional Information

In Svelte, props without an assigned value are considered as a _required_ prop. The example above might work as well since it's in JavaScript. However, consider the next example in Typescript.

```tsx
import { useDefaultProps } from '@builder.io/mitosis';

interface MyComponentProps {
  name?: string;
}

export default function MyComponent(props: MyComponentProps) {
  useDefaultProps<MyComponentProps>({
    name: undefined,
  });

  return (
    <div>{props.name}</div>
  );
}
```

which yields:

```svelte
<script context="module" lang="ts">
  interface MyComponentProps {
    name?: string;
  }
</script>

<script lang="ts">
  export let name: MyComponentProps["name"];
</script>

<div>{name}</div>
```

Which apparently is not enough for TypeScript (and/or Svelte?). It needs explicit `undefined` (as follow) so that it will stop saying that "Property 'name' is missing".

```svelte
<script context="module" lang="ts">
  interface MyComponentProps {
    name?: string;
  }
</script>

<script lang="ts">
  export let name: MyComponentProps["name"] = undefined;
</script>

<div>{name}</div>
```

This issue might need additional check for other frameworks since I'm not familiar with them.

### PR DESCRIPTION
**Pull request by kingzez**
## Description

Add a short description of:
- what changes you made, 
- why you made them, and 
- any other context that you think might be helpful for someone to better understand what is contained in this pull request. 

This sort of information is useful for people reviewing the code, as well as anyone from the future trying to understand why changes were made or why a bug started happening.

Fix https://github.com/BuilderIO/mitosis/issues/847
Fix #913

### CODE REVIEW

File: packages/core/src/__tests__/__snapshots__/parse-jsx.test.ts.snap [hunk 1]
```diff
@@ -7674,9 +7702,22 @@ exports[`Parse JSX > defaultPropsOutsideComponent 1`] = `
...
   "defaultProps": {
-    "link": "https://builder.io/",
-    "openLinkInNewTab": false,
-    "text": "default text",
+    "link": {
+      "code": "https://builder.io/",
+      "type": "property",
+    },
+    "onClick": {
+      "code": "() => {",
```
Comment (samijaber) [inline packages/core/src/__tests__/__snapshots__/parse-jsx.test.ts.snap:7710]: This doesn't look right...it's missing `}` from the end 😮

**Review by samijaber — CHANGES_REQUESTED** [review]
Amazing, thank you for this PR!

Just need to fix the issue with the trailing `}` being removed in the JSX parser.

File: packages/core/src/parsers/jsx/function-parser.ts [hunk 2]
```diff
@@ -20,6 +20,21 @@ export function generateUseStyleCode(expression: babel.types.CallExpression) {
...
 
+function processDefaultPropsValue(value: any) {
+  if (types.isFunctionExpression(value) || types.isArrowFunctionExpression(value)) {
+    const code = generate(value)
+      .code.trim()
+      // Remove arbitrary block wrapping if any
+      // AKA
+      //  { console.log('hi') } -> console.log('hi')
+      .replace(/^{/, '')
+      .replace(/}$/, '');
```
Comment (kingzez) [inline packages/core/src/parsers/jsx/function-parser.ts:31]: @samijaber  I  trailing } here, I follow your [issues tips](https://github.com/BuilderIO/mitosis/issues/847#issuecomment-1307439504), I thought it should be done that way,  I change it later ^-^

Comment (samijaber) [reply packages/core/src/parsers/jsx/function-parser.ts]: Yup, sorry for that! Mistake on my part

Comment (kingzez) [reply packages/core/src/parsers/jsx/function-parser.ts]: Hahaha，it's my fault

File: packages/core/src/parsers/jsx/function-parser.ts [hunk 3]
```diff
@@ -32,7 +41,7 @@ export function parseDefaultPropsHook(
       if (i.key?.name) {
         component.defaultProps = {
           ...(component.defaultProps ?? {}),
-          [i.key?.name]: i.value.value,
+          [i.key?.name]: processDefaultPropsValue(i.value),
```
Comment (samijaber) [inline packages/core/src/parsers/jsx/function-parser.ts:44]: Actually, I recently merged logic that handles all of this logic cleanly: `parseStateObjectToMitosisState`

You can simplify this function and write:

```ts
  if (types.isObjectExpression(firstArg)) {
    component.defaultProps = parseStateObjectToMitosisState(firstArg)
  }
```

If you don't mind doing that, would be amazing 🙏🏽

Comment (kingzez) [reply packages/core/src/parsers/jsx/function-parser.ts:44]: I tried, but change to `parseStateObjectToMitosisState `, the snapshot mismatched,  property `code` has extra double quote and function with name

```js
 FAIL  src/__tests__/parse-jsx.test.ts > Parse JSX > defaultProps
AssertionError: Snapshot `Parse JSX > defaultProps 1` mismatched
  - Expected  - 4
  + Received  + 4

        set": {},
      },
      "defaultProps": {
        "link": {
  -       "code": "https://builder.io/",
  +       "code": "'https://builder.io/'",
          "type": "property",
        },
        "onClick": {
  -       "code": "() => {
  +       "code": "onClick() {
      console.log('hi');
    }",
          "type": "method",
        },
        "openLinkInNewTab": {
  -       "code": false,
  +       "code": "false",
          "type": "property",
        },
        "text": {
  -       "code": "default text",
  +       "code": "'default text'",
          "type": "property",
        },
      },
      "exports": {},


```

Comment (samijaber) [reply packages/core/src/parsers/jsx/function-parser.ts:44]: Oh, right! Sorry for the confusion. We recently changed this part of our JSON schema, during the time you made your PR. 

We actually no longer need to `JSON.stringify` the output in the generators! So everything that you do this:

```ts
return `${prop}: ${isMethod ? `${value}` : json5.stringify(value)}`;
```

You can now replace with just:

```ts
return `${prop}: ${value}`;
```

the `parse-jsx` snapshot are going to need to be updated, but the final code snapshots in React/Vue/etc. will remain the same.

Sorry for sending you on a long journey with this PR: I forgot to tell you all this information at the very start. Thank you for your patience 🙏🏽

Comment (kingzez) [reply packages/core/src/parsers/jsx/function-parser.ts:44]: Double quotes can solve it, but function code get `onClick() {
      console.log('hi');
    }`,  the defaultProps's function should not has function name, it should be `() => {
      console.log('hi');
    }`, should not converted

Is defaultProps that should handled as state? 🤔

Comment (kingzez) [reply packages/core/src/parsers/jsx/function-parser.ts:44]: https://github.com/BuilderIO/mitosis/blob/main/packages/core/src/parsers/jsx/state.ts#L108-L124

```ts
if (types.isFunctionExpression(item.value)) {
  return {
    code: parseCode(item.value),
    type: 'function',
  };
} else if (types.isArrowFunctionExpression(item.value)) {
  const n = babel.types.objectMethod(
    'method',
    item.key as babel.types.Expression,
    item.value.params,
    item.value.body as babel.types.BlockStatement,
  );
  const code = parseCode(n);
  return {
    code: code,
    type: 'method',
  };
}
```

I see this part is dealing with that PR https://github.com/BuilderIO/mitosis/pull/875

**Review by samijaber — APPROVED** [review]
Amazing @kingzez , thank you for this PR 😄

**Pull request merged**